### PR TITLE
Add a BlockBasedTableOption to turn off index block compression.

### PR DESCRIPTION
**Pull request by anand1976**
Summary:
Add a new bool option index_uncompressed in BlockBasedTableOptions.

Test Plan:
1. New unit test in table_test.cc
2. dbbench to check size w/wo index compression -

  ./db_bench --benchmarks="fillseq,stats" -enable_index_compression=true
  ls -l /tmp/rocksdbtest-142225/dbbench/*.sst
    -rw-r--r--. 1 anand76 users 30287419 Dec 20 21:46
    /tmp/rocksdbtest-142225/dbbench/000007.sst

  ./db_bench --benchmarks="fillseq,stats" -enable_index_compression=false
  ls -l /tmp/rocksdbtest-142225/dbbench/*.sst
    -rw-r--r--. 1 anand76 users 30470586 Dec 20 21:49
    /tmp/rocksdbtest-142225/dbbench/000007.sst

### CODE REVIEW

File: table/block_based_table_builder.cc [hunk 1]
```diff
@@ -784,17 +784,27 @@ Status BlockBasedTableBuilder::Finish() {
...
     while (s.IsIncomplete()) {
       s = r->index_builder->Finish(&index_blocks, index_block_handle);
       if (!s.ok() && !s.IsIncomplete()) {
         return s;
       }
-      WriteBlock(index_blocks.index_block_contents, &index_block_handle,
-                 !is_data_block);
+      if (!r->table_options.index_uncompressed) {
+        WriteBlock(index_blocks.index_block_contents, &index_block_handle,
+                   !is_data_block);
```
Comment (ajkr) [inline table/block_based_table_builder.cc]: we should get rid of this variable and just pass false

**Comment by anand1976** [comment]
Is it a widely held opinion? It seems a little counter intuitive to take expected behavior (i.e compress index blocks using the same compression type as data blocks) and turn that into an option, albeit one that's enabled by default. Having an option to avoid index compression seems more intuitive.

File: table/block_based_table_builder.cc (hunk 1 above)
Comment (anand1976) [reply table/block_based_table_builder.cc]: Agreed. Not sure why the code had this variable.

**Comment by siying** [comment]
@anand1976 the opinion comes from @mcallahan to avoid double negative in some setting. "enable_compression" is better than "disable_compression", because it's easier to mistake when setting "disable_compression=false".

**Comment by ajkr** [comment]
I'm not sure. FWIW, the default behavior wasn't that expected to me (I didn't know until now) since we already pick shortest boundary keys to store in index block and delta-encode them, so whether compression is worthwhile seems non-obvious.

**Comment by anand1976** [comment]
Thanks @siying and @ajkr for the clarification. I'll change the option name to enable_index_compression. Default value will be true.

**Pull request closed without merging**